### DISCUSSION TITLE
Use of a LLM for "About me" text

### DISCUSSION THREAD

**Discussion by wintermeyer** [wintermeyer/animina#390] [Ideas]
I think it would be a huge help for many users to get assistance of an inbuilt LLM to e.g. generate an initial "About me" story. We could pipe the white, green and red flags plus the data we have (e.g. age, occupation, city) into it and generate a funny and a serious version. It will never be as good as written by a human but for 80% of users it will be a very good start.

## Questions

- Which LLM would be the best fit and is available with an open-source license?
- How long would it take to implement it? Ballpark.
- Is the LLM good enough for being used with German. We should try this first.

CC: @briankariuki

**Comment by briankariuki** [comment]
Llama would be a good LLM to use.

Bumblebee supports llama models so we have to find one that has been fintuned on german text from huggingface.

**Reply by wintermeyer** [reply]
> Llama would be a good LLM to use.

Please help me understand the needed resources for this. Would we have to load more than 10 GB of data every time we start the LLM because of the size of these files?

<img width="720" alt="Bildschirmfoto 2024-05-03 um 16 19 44" src="https://github.com/animina-dating/animina/assets/69770/eae9dfb9-3de7-4795-ae2d-dc1a7a163edc">

Or would we load that once at the start of the application?

**Reply by briankariuki** [reply]
Ideally the model will be loaded on application startup and occupy the 10gb ram.

I would recommend using a machine with a gpu instead of cpu. On cpu inference might take too long.

Another option would be to use a 3rd party service something like replicate/groq via an api.

**Comment by briankariuki** [comment]
This should take about 2 to 3 days to implement

**Comment by briankariuki** [comment]
A few models I've found

https://huggingface.co/flozi00/Llama-2-7b-german-assistant-v3

https://huggingface.co/flozi00/Mistral-7B-german-assistant-v5

https://huggingface.co/flozi00/Llama-2-13b-german-assistant-v4

**Comment by wintermeyer** [comment]
This has been implimented.
